### DISCUSSION TITLE
Can I connect VScode to a local code editor?

### DISCUSSION THREAD

**Discussion by Codrineye** [microsoft/vscode-discussions#1045] [Extension Development QnA] [answered]
I'm a student and I code in C++, for the past 3 years I've been trying, on and off, to configure the C++ extension to work as I expect (eg: not giving me include errors on an extension generated test).
I use Code::Blocks for my C++ needs, is there a way to connect VScode to this IDE?

**Comment by gjsjohnmurray** [comment, marked as answer]
This is not the right forum for your question. Please review the [READ FIRST](https://github.com/microsoft/vscode-discussions/discussions/5) post pinned at the top of the forum, as it explains why not and gives links to other resources you could use.

**Reply by Codrineye** [reply]
Sorry
I saw "ask questions" and thought I could finally get help

**Comment by abrilurena** [comment]
Hi, @Codrineye! I hope you're doing great. I found this, please let me know if it helps ✨ You can set up your C++ environment in VS Code to achieve a smooth development experience. Let’s walk through the steps:


**1. Install the C/C++ Extension for VS Code:**
  - The C/C++ extension for Visual Studio Code provides features like syntax highlighting, IntelliSense, and error checking.
  - Open VS Code, go to the Extensions view (click the icon on the Activity bar or use the shortcut ⇧⌘X on macOS or Ctrl+Shift+X on Windows/Linux).
  - Search for ‘C++’ and install the extension.
  
**2. Set Up Your C++ Environment:**
      - C++ is a compiled language, so your source code needs to be translated (compiled) before running.
      - The C/C++ extension doesn’t include a compiler or debugger; it relies on command-line tools.
      - Check if you already have a C++ compiler installed:
      - Open a new VS Code terminal window using ⌃⇧ (Windows/Linux Ctrl+Shift+).
      - Run the following command to check for the GCC compiler (g++):
            `g++ --version`

  If neither GCC nor Clang is found, install a compiler (see next step).

**3. Install a Compiler:**
- Choose a compiler based on your platform:
- Windows:
    - You can install MinGW-w64 via MSYS2. It provides native builds of GCC and other C++ tools.
    - Download the MinGW installer and follow the installation steps.
- Linux:
    - Commonly preinstalled compilers include the GNU Compiler Collection (GCC).
    - If not installed, follow the [GCC tutorial](https://code.visualstudio.com/docs/languages/cpp).
- macOS:
    - Clang tools come with Xcode.
    - If not installed, follow the [Clang tutorial](https://code.visualstudio.com/docs/languages/cpp).

**4. Configure VS Code for C++ Development:**
- Set up your c_cpp_properties.json file to specify include paths and compiler options.
- Create a new folder or open an existing one in VS Code.
- Write your C++ code, and use the C/C++ extension features.

**5. Additional Notes:**
- If you prefer a full IDE experience, consider using Visual Studio Community edition.

**Reply by Codrineye** [reply]
Hello.
Thank you for coming to help, I have recieved help on StackOverflow with this [The Fix](https://stackoverflow.com/q/77943103/20438012).
I have attempted the GCC tutorial multiple times, never getting expected behaviour.
I did try VS Community Edition many times, however, the IDE too gave me build errors I had never encountered on Code::Blocks.

I'd still like to thank you, once again, for attempting to help. That approach still ends in errors on my part tho.

**Comment by Codrineye** [comment]
I asked out of desperation without checking if this was the appropiate place for it.
Feel free to ignore this.

There is no need to connect to an IDE to VSCode, access [THIS STACKOVERFLOW](https://stackoverflow.com/q/77943103/20438012) if you have include errors when compiling with the C/C++ extension.

**Comment by abrilurena** [comment]
Hi, @Codrineye! I hope you're doing great. I know it's been a while, but i just wanted to let you know that  in a future, if you have any questions/need help with something - please feel free to reach out, will do our best to help you ✨
